docs/source/en/tutorials/custom_pipelines_components.md

```diff
@@ -0,0 +1,135 @@
+
+<Tip warning={true}>
+
+You should use `trust_remote_code=True` _only_ when you fully trust the code and have verified its usage.
+
+</Tip>
+
+## Pipeline components
+
+In the interest of brevity, you'll use the custom UNet, scheduler, and pipeline classes that we've already authored:
+
+```bash
+# Custom UNet
+wget https://huggingface.co/sayakpaul/custom_pipeline_remote_code/raw/main/unet/my_unet_model.py
+# Custom scheduler
+wget https://huggingface.co/sayakpaul/custom_pipeline_remote_code/raw/main/scheduler/my_scheduler.py
+# Custom pipeline
+wget https://huggingface.co/sayakpaul/custom_pipeline_remote_code/raw/main/my_pipeline.py
+```
+
+<Tip warning={true}>
+
+The above classes are just for references. We encourage you to experiment with these classes for desired customizations.
+
+</Tip>
+
+Load the individual components, starting with the UNet:
+
+```python
+from my_unet_model import MyUNetModel
+
+pretrained_id = "hf-internal-testing/tiny-sdxl-custom-all"
+unet = MyUNetModel.from_pretrained(pretrained_id, subfolder="unet")
+```
+
+Then go for the scheduler:
+
+```python
+from my_scheduler import MyUNetModel
+
+scheduler = MyScheduler.from_pretrained(pretrained_id, subfolder="scheduler")
+```
+
+Finally, the VAE and the text encoders:
+
+```python
+from transformers import CLIPTextModel, CLIPTextModelWithProjection, CLIPTokenizer
+from diffusers import AutoencoderKL
+
+text_encoder = CLIPTextModel.from_pretrained(pretrained_id, subfolder="text_encoder")
+text_encoder_2 = CLIPTextModelWithProjection.from_pretrained(pretrained_id, subfolder="text_encoder_2")
+tokenizer = CLIPTokenizer.from_pretrained(pretrained_id, subfolder="tokenizer")
+tokenizer_2 = CLIPTokenizer.from_pretrained(pretrained_id, subfolder="tokenizer_2")
+
+vae = AutoencoderKL.from_pretrained(pretrained_id, subfolder="vae")
+```
+
+`MyUNetModel`, `MyScheduler`, and `MyPipeline` use blocks that are already supported by Diffusers. If you are using any custom blocks make sure to put them in the module files themselves. 
+
+## Pipeline initialization and serialization
+
+With all the components, you can now initialize the custom pipeline:
+
+```python
+pipeline = MyPipeline(
+    vae=vae, 
+    unet=unet, 
+    text_encoder=text_encoder, 
+    text_encoder_2=text_encoder_2,
+    tokenizer=tokenizer, 
+    tokenizer_2=tokenizer_2, 
+    scheduler=scheduler,
+)
+```
+
+Now, push the pipeline to the Hub:
+
+```python
+pipeline.push_to_hub("custom_pipeline_remote_code")
+```
+
+Since the `pipeline` itself is a custom pipeline, its corresponding Python module will also be pushed ([example](https://huggingface.co/sayakpaul/custom_pipeline_remote_code/blob/main/my_pipeline.py)). If the pipeline has any other custom components, they will be pushed as well ([UNet](https://huggingface.co/sayakpaul/custom_pipeline_remote_code/blob/main/unet/my_unet_model.py), [scheduler](https://huggingface.co/sayakpaul/custom_pipeline_remote_code/blob/main/scheduler/my_scheduler.py)). 
```

Comment:
```suggestion
Since the `pipeline` itself is a community pipeline, its corresponding [Python module](https://huggingface.co/sayakpaul/custom_pipeline_remote_code/blob/main/my_pipeline.py) is also pushed. If the pipeline has any other custom components, they are pushed as well ([UNet](https://huggingface.co/sayakpaul/custom_pipeline_remote_code/blob/main/unet/my_unet_model.py), [scheduler](https://huggingface.co/sayakpaul/custom_pipeline_remote_code/blob/main/scheduler/my_scheduler.py)). 
```